app/models/fe/concerns/answer_sheet_concern.rb

```diff
@@ -69,8 +69,8 @@ def reference?
       false
     end
 
-    def percent_complete
-      num_questions = question_sheets.inject(0.0) { |sum, qs| qs.nil? ? sum : qs.questions_count + sum }
+    def percent_complete(required_only = false)
```

Comment:
I realize this would be considered a backwards incompatible change, but I think the intention was probably always to have percent_complete based on required only. I'd prefer defaulting this to `true`.